src/app/Fake.IO.FileSystem/ChangeWatcher.fs

```diff
@@ -0,0 +1,112 @@
+[<AutoOpen>]
+/// This module contains helpers to react to file system events.
+module Fake.IO.FileSystem.ChangeWatcher
+
+open System.IO
+open Fake.Core
+open Fake.IO
+
+type FileStatus =
+    | Deleted
+    | Created
+    | Changed
+
+type FileChange =
+    { FullPath : string
+      Name : string
+      Status : FileStatus }
+
+type WatchChangesOption =
+    { IncludeSubdirectories: bool }
+
+let private handleWatcherEvents (status : FileStatus) (onChange : FileChange -> unit) (e : FileSystemEventArgs) =
+    onChange ({ FullPath = e.FullPath
+                Name = e.Name
+                Status = status })
+
+
+/// Watches the for changes in the matching files.
+/// Returns an IDisposable which allows to dispose all FileSystemWatchers.
+///
+/// ## Parameters
+///  - `onChange` - function to call when a change is detected.
+///  - `fileIncludes` - The glob pattern for files to watch for changes.
+///
+/// ## Sample
+///
+///     Target "Watch" (fun _ ->
```

Comment:
We could update that comment to represent new APIs. (For example `Target.Create`, or we leave out the Targets module completely)